app.rb

```diff
@@ -1,128 +1,67 @@
-require './nameable'
-require './person'
-require './teacher'
-require './student'
-require './classroom'
-require './book'
-require './rental'
+require './apps/books_app'
+require './apps/persons_app'
+require './apps/rentals_app'
+require './classes/nameable'
+require './classes/person'
+require './classes/teacher'
+require './classes/student'
+require './classes/classroom'
+require './classes/book'
+require './classes/rental'
 
 class App
-  def initialize
-    @books = []
-    @persons = []
-    @rentals = []
-  end
+  attr_accessor :books, :persons
 
-  def start_console
-    puts '********** Welcome to the school library app **********'
-    until list_options
-      input = gets.chomp
-      option(input)
-    end
+  def initialize
+    @books = BooksApp.new
+    @persons = PersonApp.new
+    @rentals = RentalsApp.new(@books.books, @persons.persons)
   end
 
-  def list_books
-    puts 'there is no book in the library' if @books.empty?
-    @books.each_with_index do |book, index|
-      puts " [Book]- Index: #{index},  Title: #{book.title}, Author: #{book.author}, Id: #{book.id}"
-    end
+  def list_options
+    puts
+    puts ' Please choose among the options below by pressing its corresponding number:'
+    puts '[1] List all books'
+    puts '[2] List all people'
+    puts '[3] Add a person'
+    puts '[4] Add a book'
+    puts '[5] Add a rental'
+    puts '[6] List all rentals for a given person id'
+    puts '[7] Exit'
   end
 
-  def list_persons
-    puts 'there is no person registred in the library' if @persons.empty?
-    @persons.each_with_index do |person, index|
-      puts "[#{person.class.name}] Index: #{index}, Name: #{person.name}, Age: #{person.age}, id: #{person.id} "
-    end
+  def exit_msg
+    puts ' '
+    puts 'Thank you for using this App'
+    exit(true)
   end
 
-  def add_person
-    print 'To create a student, press 1, to create a teacher, press 2 : '
-    option = gets.chomp
-    case option
+  def option(input)
+    case input
     when '1'
-      create_student
+      @books.list_books
     when '2'
-      create_teacher
-    else
-      puts 'Invalid input. Try again'
-    end
-  end
-
-  def create_student
-    puts '********** Create a new student **********'
-    print 'Enter student name: '
-    name = gets.chomp
-    print 'Enter student age: '
-    age = gets.chomp.to_i
-    print 'Has parent permission? [Y/N]: '
-    parent_permission = gets.chomp.downcase
-    case parent_permission
-    when 'y'
-      classroom = 'classroom A'
-      student = Student.new(age, classroom, name, parent_permission: true)
-    when 'n'
-      classroom = 'classroom B'
-      student = Student.new(age, classroom, name, parent_permission: false)
+      @persons.list_persons
+    when '3'
+      @persons.add_person
+    when '4'
+      @books.add_book
+    when '5'
+      @rentals.add_rental
+    when '6'
+      list_filtred_rentals
```

Comment:
- [ ] **Required:** When I tried to view the rentals before exiting the app, it crashes. Kindly look into this behaviour.

Reply:
Thank you @brytebee for your review, I have made the required changes.